### DISCUSSION TITLE
setpoints in crazyflie client software

### DISCUSSION THREAD

**Discussion by adarsh2798** [bitcraze/discussions#1255] [Q&A] [answered]
Hi
I want to understand what setpoints crayflie receives when we control crazyflie using ``command based flight control`` tab in client software. Like, What does takeoff, up,down etc buttons mean? Do they vary based on controller set using parameter tab? Can Someone list the setpoints for different conroller like PID, brescianini, mellinger?

**Comment by gemenerik** [comment, marked as answer]
It uses the high level commander. Checkout the [_flight_command method of the cfclient](https://github.com/bitcraze/crazyflie-clients-python/blob/master/src/cfclient/ui/tabs/FlightTab.py?plain=1#L258-L280) for the exact commands per button. This can be used with whatever controller you have configured.
